[WC-439] Add docs for the new barcode scanner widget (#3297)
mendix/docs: content/appstore/widgets/barcode-scanner.md

---
title: "Barcode scanner"
category: "Widgets"
description: "Describes the configuration and usage of the Barcode scanner widget, which is available in the Mendix Marketplace."

> tags: ["marketplace", "marketplace component", "widget", "barcode scanner", "phonegap", "platform support"]

Reviewer: Will this also work in phonegap or should this tag be removed?

Author: nice catch, this was leftover from previous native implementation. will remove 😄

---

## 1 Introduction

The [Barcode scanner](https://appstore.home.mendix.com/link/app/1469/) widget enables a barcode scanning functionality within your Mendix web application.

The widget does the following:

- Displays a fullscreen canvas overlay that allows the user to scan a barcode.
- Also supports QR codes and a wide range of different barcode types.
- Allows additional styling customizations through Atlas.

{{% alert type="info" %}}
For security reasons, certain browsers will only allow access to an user device's camera over a HTTPS connection and not a default HTTP connection.
Since this widget accesses the user device's camera through this method, the same constraints apply for using this widget in both production and local development.
For production, an appropriate security setting can be applied when deploying to the Mendix cloud.
For local development, the easiest solution is to set up a local proxy that secures the connection towards your local Mendix application.
{{% /alert %}}

## 2 Usage

The only requirement for the Barcode scanner widget is that it needs to be placed inside a data view that provides an object with a string attribute.

> This attribute will be used to store the resulting string of the barcode detection in.

Reviewer: I find this sentence difficult to read. I suggest, "The barcode detection result, being a string, will be committed to this attribute."

When the widget is placed on a page, it will automatically put a black canvas overlay on top of the entire page.
This canvas will provide users the ability to see their device's camera feed and scan barcodes if they allow the permission request from the application.

### 2.1 Data Source Tab

> - **Barcode** - the attribute on the data view object in which the string result of the barcode detection will be stored in.

Reviewer: If you agree with my above suggestion, this could also be reworded.

  This widget property is required.

### 2.2 Mask Tab

- **Show barcode mask** - Whether the black canvas overlay with a focus square is shown on top of the user's camera feed.
  If this property is disabled, then the entire screen will be the user's camera feed when the barcode scanner is open.

>   This property is by defauled enabled.

Reviewer: by default*

## 2 Usage

The only requirement for the Barcode scanner widget is that it needs to be placed inside a data view that provides an object with a string attribute.
The barcode detection result, being a string, will be stored in this attribute.

When the widget is placed on a page, it will automatically put a black canvas overlay on top of the entire page.
This canvas will provide users the ability to see their device's camera feed and scan barcodes if they allow the permission request from the application.

### 2.1 Data Source Tab

> - **Barcode** - the attribute on the data view object in which the barcode detection, being a string, will be stored in.

Reviewer: Apologies, I should have been more clear. I suggest, "the attribute of the Data view object which will store the barcode detection result, which is a string."

> - **Barcode** - the attribute on the data view object in which the barcode detection, being a string, will be stored in.

Author: ```suggestion
- **Barcode** - the attribute of the data view object which will store the barcode detection result, which is a string.
```
